internal/cmd/cli/apply/apply.go

```diff
@@ -331,30 +344,62 @@ func saveOCIBundle(ctx context.Context, c *client.Client, bundle *fleet.Bundle,
 	if err != nil {
 		return err
 	}
-	// OCI artifact stored successfully
-	// We don't store the resources in the bundle. Just keep the manifestID for
-	// being able to access the bundle's contents later.
-	bundle.Spec.Resources = nil
-	bundle.Spec.ContentsID = manifestID
+	logrus.Infof("OCI artifact stored successful: %s %s", bundle.Name, manifestID)
 
-	bundleCreated, err := c.Fleet.Bundle().Create(bundle)
-	if err != nil {
+	obj, err := c.Fleet.Bundle().Get(bundle.Namespace, bundle.Name, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	// when using the OCI registry manifestID won't be empty
-	// In this case we need to create a secret to store the
-	// OCI registry reference and credentials so the fleet controller is
-	// able to access.
-	secret := &corev1.Secret{
+
+	if apierrors.IsNotFound(err) {
+		// We don't store the resources in the bundle. Just keep the manifestID for
+		// being able to access the bundle's contents later.
+		bundle.Spec.Resources = nil
+		bundle.Spec.ContentsID = manifestID
+		obj, err = c.Fleet.Bundle().Create(bundle)
+		if err != nil {
+			return err
+		}
+
+		if err := createOrUpdate(c, newOCISecret(manifestID, obj, opts)); err != nil {
```

Comment:
I wonder if at some point we might want to make this transactional: if creating the secret fails, the bundle will still have been created, but would be useless. We could (eventually) try to roll the process back by deleting the bundle before returning an error.

Reply:
Not sure, since we do not have locking. If an error occurs the reconcile will be retried and should pick up where the previous one failed. Ideally the reconciles are idempotent.